### ISSUE TITLE
Add predefined filters to the top of the Templates tables

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#955]
Once #815 is complete, Add 3 predefined filters above the funder and normal template tables on the templates page. The following filters should be clickable links that provide a record count and apply only to the specific table (similar to the screenshot below):
- link text: 'All (%{nbr_templates})', behavior: reloads the table removing any search criteria
- link text:'Published (%{nbr_templates})', behavior: fetches only the published templates removing any previous search criteria
- link text:'Unpublished (%{nbr_templates})', behavior: fetches only the unpublished templates removing any previous search criteria
![screen shot 2017-12-05 at 11 48 34 am](https://user-images.githubusercontent.com/12703638/33627417-4db649a6-d9b2-11e7-8ad4-40c06930b89e.png)

**Comment by briri** [comment]
All 3 predefined filters/scopes that work with pagination:
![screen shot 2017-12-19 at 11 39 17 am](https://user-images.githubusercontent.com/1204467/34175453-5440cda0-e4b1-11e7-82b0-71f1f361e45c.png)

**Comment by stephaniesimms** [comment]
the filters are great, but some behavior is wonky. first general comments:
- it would be helpful to highlight the selected filter somehow, either make it bold or retain the gray box when I click one of the 3 (All, Published, Unpublished)
- we also need to clarify somehow that for Org Admins "Published/Unpublished" refers to their customizations on a funder template as opposed to the template status itself (screenshot). @sjDCC thoughts?

![screen shot 2018-01-02 at 3 52 14 pm](https://user-images.githubusercontent.com/12703638/34504688-f37c7c56-efd4-11e7-9727-07edf0ff8f66.png)

things that don't work as expected:
- when I click "Published" in both tabs as a Super Admin I get the expected list, but if I click the View all link to expand the list it seems to revert back to displaying All templates
- as an Admin I should see 1 Published template (screenshot) for Univ of Edinburgh but it doesn't appear in the count or when I click "Published"
![screen shot 2018-01-02 at 3 45 29 pm](https://user-images.githubusercontent.com/12703638/34504575-275f175a-efd4-11e7-9670-0db018d2679e.png)


- when I click "Unpublished" in both tabs I get a mixed list that appears to contain All templates, with published and unpublished things. It's a different list from "All" but I'm not sure what the logic is for displaying things here.

- the counts are weird/incorrect for Funder templates (screenshot - there are obviously Published templates but the count is "0"). the numbers should be calculated regardless of my affiliation as a Super Admin.
![funder-zero-published](https://user-images.githubusercontent.com/12703638/34504640-a0b72de0-efd4-11e7-8e3e-14cbcdb6e73a.png)

**Comment by sjDCC** [comment]
Hi @stephaniesimms @briri. I don't remember discussing adding this functionality and I'm not sure that we should as it's a bit confusing and the validations seem buggy currently. In all other tables, the filters re-order what is in any given column or allow a search. Why are we doing something different here by adding additional published filters above? Would we not just be best to use the status column and search?

It seems that published templates are often being recorded as unpublished - e.g. all 18 funder templates showing as unpublished despite the status in the table saying 'published'.  I'm also unclear where the status "Funder version is un-published" in Steph's screengrab 1 comes from. I don't recall ever seeing this label before. Is it new? It seems counter-intuitive to let an org customise an unpublished template so perhaps it's better to remove it from the table view when unpublished instead? Unless this is to give institutions access to a legacy customisation?

@stephaniesimms asks "we also need to clarify somehow that for Org Admins "Published/Unpublished" refers to their customizations on a funder template as opposed to the template status itself (screenshot). @sjDCC thoughts?" How did you get this view? I can't replicate it on my account so perhaps I need a secondary one that doesn't have super-admin privileges so I can view things as a standard org admin? Looking at DMPonline, it seems the 3 statuses we have currently are:
- "Published"
- "Unpublished"
- "Original funder template has changed!" (this is linked to the action 'Transfer customisation')
Perhaps we also have an "Unpublished changes" one? 
I think the easiest switch language wise (if the table accommodates it) is adding the word customisation i.e. "Customisation published" and "Customisation unpublished" Alternatively, we could have a note at the top of this table view denoting that the status refers to the institutional customisation, as all funder templates are published.

Given that the main utility of this feature is for super-admins with hundreds of templates to navigate, I'd focus on it post MVP and ensure the main features are ready for release instead. For standard admin users, they're never really having to browse through too many templates so shouldn't need this.

**Comment by briri** [comment]
Hey @sjDCC and @stephaniesimms,

The filters are indeed a bit confusing (especially on the customizable tab) due in a large part to the current user's permission level. If the user is an Org Admin what constitutes as 'published'? Is it the customizations of those funder templates? Then what counts as published if the user is a super admin. The addition of the 'all templates' tab in #900 helps reduce the confusion there since the customizable table is no longer doing double duty. 

I would suggest two options. A) wait until #900 has ben reviewed and deployed to see if the addition of the filters makes more sense (perhaps its only valuable on the all templates tab for example). B) Remove the filters and wait for the work being done on #815 to fix the sort/search functionality which will at least allow the results to be sorted by status thus grouping the published/unpublished templates together. Let us know how you want to proceed.

The "Funder version is un-published" status is indeed new, and deals with the scenario when a funder has no published version of a template (changing the publication status doesn't create a new version). If the org has already customized that template they would see this new status. This will be clearer once the 'All templates' tab is deployed.

**Comment by sjDCC** [comment]
Sounds good @briri Probably best for work to be done on both #900 and #815 then review what makes most sense here. If we can avoid having another way of filtering I'd prefer it for usability and consistency reasons, but being able to restrict the view to only published templates may be useful.

I agree that the 'all templates' tab in #900 should help. My understanding is that the status on the 'Customizable tab' should refer to the specific unis customisations, but it does raise a question of what status a super-admin sees. I guess we'll always see this page as a defined org or funder due the affiliation switcher. I was trying to figure out earlier what org I'd need to select to view pages as a super-admin, but I'm not sure there is one as you always need to edit as a specific org so that new content inherits this affiliation.

Thanks for clarifying the new "Funder version is unpublished" status. That makes sense if it's a legacy customisation as we don't want admins to lose access to any content they've added. It should probably always be superseded by "Original funder template has changed!" as they're likely to release a new version instead of unpublishing their templates completely, but good to have in case of that eventuality.

**Comment by stephaniesimms** [comment]
@sjDCC this was meant to be a simple usability improvement for the templates table - aimed at Super Admins but also DMPTool Org Admins who have to navigate 43+ funder templates. I didn't anticipate there would be so many complicating factors - my apologies for taking us down this rabbit hole - but I still think it's important for users to be able to quickly identify total numbers of templates and which are Published. I need as many cues as possible to remind me to take the action to publish. 

many thanks to @briri and @xsrust for devising this approach. 
All Templates - Org Templates - Customizable Templates tabs for Super Admins is much easier for editing funder templates and creates more consistency between Org Admin and Super Admin template functionality. There does seem to be a persistent issue with the counts reflecting what is actually published or not (see screenshot).

![screen shot 2018-01-03 at 1 45 50 pm](https://user-images.githubusercontent.com/12703638/34541378-d2856a18-f08c-11e7-80af-2fa5f1105bb3.png)

**Comment by briri** [comment]
The new 3 tab view is out on roadmap-stg @sjDCC and @stephaniesimms. Please use this ticket to for any changes that should be made.

**Comment by sjDCC** [comment]
Agree with @stephaniesimms that we need to modify the language on the Customisable Templates tab to make clear that what is (un)published is the customisation not the template. 

It would be a very rare case when we present an unpublished funder template to the org-admins i.e. only if the funder unpublishes their template completely and doesn't replace it with a new version. 

When a new version of a funder template is released the customisation published status moves to "Original funder template has changed!" Effectively the customisation is still published and valid for any existing plans created with the old version of the funder template, but it won't be visible on the new version of the funder template until the user transfers it, check/edits as needed and republishes. See explanation of workflow in [user guide ](https://github.com/DMPRoadmap/roadmap/wiki/User-guide#transfer-a-customisation)

**Comment by stephaniesimms** [comment]
i agree w/ @sjDCC on above points. so to summarize @briri :
- "unpublished" should never have a hyphen
- modify the language at the top of the Customisable Templates table to say "All," "Customisation published," and "Customisation unpublished."
- Super Admins shouldn't see the 'Customisable Templates' tab. Funders don't need to customize their own or others' templates and we can adjust our privileges to be Org Admins if we need to test or demo this functionality.

One remaining issue is that none of the filters at the top registers anything as "Published" - all are zero despite the tables containing published items:
![screen shot 2018-01-05 at 1 28 02 pm](https://user-images.githubusercontent.com/12703638/34629274-6c80db3a-f21c-11e7-98cc-930310bbb384.png)

**Comment by sjDCC** [comment]
Hi @stephaniesimms I didn't ask this question earlier, but how does a superadmin change their privileges to be an org admin. I assume we have to remove the additional permissions which, unless I'm missing something, would then mean we couldn't add them back on ourselves as we've effectively downgraded the account. 

I've ensured I have two accounts - one as super admin and one as org admin for test / demo purposes. I change the org affiliation on the test account via superadmin controls as changing org directly from within this account will lose admin rights and mean they need to be regranted.

I think super admins should see the 'Customisable Templates' tab but only for the org they are affiliated with via the switcher above the table. If this is set as a normal org they'll see it, if it's set as a funder they won't. I'm not sure why we'd want to lose this?

**Comment by stephaniesimms** [comment]
No user, including Super Admins, is able to change their own privileges. The solution for us is to have separate accounts as you noted above, and this is what I do currently for DMPTool. 

@briri noted that it may be easier to keep the Customizable templates tab anyway rather than introduce logic to display for Org Admin but not for Super Admin. in this case, Super Admins will be able to add customizations or change their org and edit others' customizations - I just couldn't imagine why I would ever need to do this but it sounds like it's better to keep it for the reasons you and brian raised.

**Comment by sjDCC** [comment]
The text change of un-published to unpublished seems to be fixed but all templates are still be recorded as unpublished in calculations

![capture](https://user-images.githubusercontent.com/6169940/34946067-547713c2-f9fd-11e7-8a73-d97e30ddaf03.PNG)

**Comment by stephaniesimms** [comment]
this seems to be working now. closing issue.

**Comment by sjDCC** [comment]
Yes, great to see these working now - thanks @briri 

If there's anything we need to do as part of the data migration to get this to work please flag it via a call or slack (and I guess we'd need to add instructions to migration path guidelines too)

Removing from board now
